umpost/um2netcdf.py

```diff
@@ -352,34 +345,61 @@ def process(infile, outfile, args):
             c.cell_methods = fix_cell_methods(c.cell_methods)
 
             try:
-                fix_latlon_coord(c, grid_type, dlat, dlon)
+                fix_latlon_coord(c, mv.grid_type, mv.d_lat, mv.d_lon)
             except iris.exceptions.CoordinateNotFoundError:
                 print('\nMissing lat/lon coordinates for variable (possible timeseries?)\n')
                 print(c)
                 raise Exception("Variable can not be processed")
 
-            fix_level_coord(c, z_rho, z_theta)
+            fix_level_coord(c, mv.z_rho, mv.z_theta)
 
             if do_mask:
                 # Pressure level data should be masked
-                if require_heaviside_uv(c.item_code):
-                    if heaviside_uv:
-                        apply_mask(c, heaviside_uv, args.hcrit)
-                    else:
-                        continue
-
-                if require_heaviside_t(c.item_code):
-                    if heaviside_t:
-                        apply_mask(c, heaviside_t, args.hcrit)
-                    else:
-                        continue
+                if require_heaviside_uv(c.item_code) and heaviside_uv:
+                    apply_mask(c, heaviside_uv, args.hcrit)
+
+                if require_heaviside_t(c.item_code) and heaviside_t:
+                    apply_mask(c, heaviside_t, args.hcrit)
 
             if args.verbose:
                 print(c.name(), c.item_code)
 
+            # TODO: split cubewrite ops into funcs & bring those steps into process() workflow
+            #       or a sub process workflow function (like process_mule_vars())
             cubewrite(c, sman, args.compression, args.use64bit, args.verbose)
 
 
+MuleVars = collections.namedtuple("MuleVars", "grid_type, d_lat, d_lon, z_rho, z_theta")
+
+
+def process_mule_vars(fields_file: mule.ff.FieldsFile):
+    """
+    Extract model levels with mule.
+
+    The model levels help with workflow dimension naming.
+
+    Parameters
+    ----------
+    fields_file : an open mule fields file.
+
+    Returns
+    -------
+    A MuleVars data structure.
+    """
+    if isinstance(fields_file, mule.ancil.AncilFile):
+        raise NotImplementedError('Ancillary files are currently not supported')
+
+    if mule.__version__ == "2020.01.1":
+        msg = "mule 2020.01.1 doesn't handle pathlib Paths properly"
+        raise NotImplementedError(msg)  # fail fast
```

Comment:
You could move this to an __init__ at some level to make it fail even faster, unless you wanted other aspects of the module to be available with mule 2020.01.1

Reply:
I added this error message to convert a comment to code, although I assumed `mule 2020` would be outdated by now. This is probably wrong if `mule 2020` versions are still mentioned in the `conda coecms` channel.

@MartinDix did you add the code comment about the `mule 2020` version not handling pathlibs properly?

Reply:
This is now for #70.